purchase_reception_notify/models/stock_picking.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from odoo import _, api, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.model
+    def _purchase_order_picking_confirm_message_content(
+            self, picking, purchase_dict):
+        if not purchase_dict:
+            purchase_dict = {}
+        title = _('Receipt confirmation %s') % (
+            picking.name)
+        message = '<h3>%s</h3>' % title
+        message += _('The following items '
+                     'have now been received in Incoming Shipment %s:') % (
+            picking.name)
+        message += '<ul>'
+        for purchase_line_id in purchase_dict.values():
+            message += _(
+                '<li><b>%s</b>: Received quantity %s %s</li>'
+            ) % (purchase_line_id['purchase_line'].product_id.display_name,
+                 purchase_line_id['stock_move'].product_qty,
+                 purchase_line_id['stock_move'].product_uom.name,
+                 )
+        message += '</ul>'
+        return message
+
+    @api.multi
+    def action_done(self):
+        super(StockPicking, self).action_done()
+        for picking in self:
```

Comment:
for picking in self.filtered(lambda p: picking.picking_type_id.code == 'incoming')